### ISSUE TITLE
[Question] What is the ideal timeout?

### ISSUE THREAD

**Issue by haramchang** [microsoft/playwright#16862]
Hi, I was wondering if a timeout of 20 seconds (for each test) for about 500 tests running in parallel (with 28 workers) is too short of a timeout? The tests consist of form submissions + making sure buttons/CTAs exist on a page - nothing that takes more than 20 seconds each. When running one test locally, it's super fast (less than 20 seconds) which is why we have our timeouts set at 20 seconds, but when everything is run on gitlab, all of them fail due to test timeouts. Is that expected because there are so many tests, or are we setting up our tests wrong?

**Comment by yury-s** [comment]
There is no universal solution unfortunately, the timeout will depend on the tests, how much times they normally need and also on how fast the CI machines are, how good the network connection between the CI and your server etc. Pick some reasonable number which is sufficient for the tests to pass. You can have different configuration on CI and locally using `process.env.CI` variable, similar to the example in [this doc](https://playwright.dev/docs/test-configuration#testing-options).

> for about 500 tests running in parallel (with 28 workers)

Machines on gitlab are proably less powerful than what you use for a local run. Make sure you don't set number of worker too high (it should normally be <= number of CPU cores and playwright can usually auto detect optimal number of parallel workers to use).

> all of them fail due to test timeouts.

There is probably something wrong with the CI environment then, try looking at [the traces](https://playwright.dev/docs/trace-viewer-intro) to see why they are timing out.

Hope this helps, if not feel free to file another issue with further questions.
